packages/stark-ui/src/modules/input-mask-directives/directives/text-mask.directive.ts

```diff
@@ -0,0 +1,112 @@
+import { Directive, ElementRef, forwardRef, Inject, Input, OnChanges, Optional, Provider, Renderer2, SimpleChanges } from "@angular/core";
+import { COMPOSITION_BUFFER_MODE, NG_VALUE_ACCESSOR } from "@angular/forms";
+import { MaskedInputDirective, TextMaskConfig as Ng2TextMaskConfig } from "angular2-text-mask";
+import { StarkTextMaskConfig } from "./text-mask-config.intf";
+
+/**
+ * Name of the directive
+ */
+const directiveName: string = "[starkTextMask]";
+
+/**
+ * @ignore
+ */
+export const STARK_TEXT_MASK_VALUE_ACCESSOR: Provider = {
+	provide: NG_VALUE_ACCESSOR,
+	// tslint:disable-next-line:no-forward-ref
+	useExisting: forwardRef(() => StarkTextMaskDirective),
+	multi: true
+};
+
+/**
+ * @ignore
+ */
+const defaultTextMaskConfig: StarkTextMaskConfig = {
+	mask: false, // by default the mask is disabled
+	guide: true,
+	placeholderChar: "_",
+	keepCharPositions: true
+};
+
+/**
+ * Directive to display a mask in input elements. This directive internally uses the {@link https://github.com/text-mask/text-mask/tree/master/core|text-mask-core} library to provide the input mask functionality.
+ *
+ * **`IMPORTANT:`** Currently the Text Mask supports only input of type text, tel, url, password, and search.
+ * Due to a limitation in browser API, other input types, such as email or number, cannot be supported.
+ *
+ * For example:
+ * ```html
+ * <input type="text" [starkTextMask]="yourMaskConfig">
+ *   // or
+ * <input type="text" [(ngModel)]="yourModelValue" [starkTextMask]="yourMaskConfig">
+ *   // or
+ * <input type="text" [formControl]="yourFormControl" [starkTextMask]="yourMaskConfig">
+ * ```
+ *
+ */
+@Directive({
+	host: {
+		"(input)": "_handleInput($event.target.value)",
+		"(blur)": "onTouched()",
+		"(compositionstart)": "_compositionStart()",
```

Comment:
I saw this on the Angular implementation. But I didn't fully understand when it is triggered. Should this also be added to the transform directive?

Reply:
Well, this is taken from the `MaskedInputDirective` from `text-mask` itself, and since we are extending from such directive, I kept everything as is... otherwise there might be some issues with this feature